packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/BaseScopeHandler.ts

```diff
@@ -104,21 +105,52 @@ export default abstract class BaseScopeHandler implements ScopeHandler {
     let previousScope: TargetScope | undefined = undefined;
     const hints: ScopeIteratorRequirements = {
       ...DEFAULT_REQUIREMENTS,
+      distalPosition:
+        requirements.distalPosition ?? direction === "forward"
+          ? editor.document.range.end
+          : editor.document.range.start,
```

Comment:
Setting distal position to be the end of the document simplifies lots of code, as that's logically what we were doing anyway